spec/widgets/auto-style/style-utils.spec.js

```diff
@@ -61,6 +61,29 @@ describe('src/widgets/auto-style/style-utils', function () {
 
       expect(StyleUtils.changeStyle(cartocss, attr, newStyle)).toBe(expected);
     });
+
+    it('should not replace line-opacity if it is included under an outline symbolizer', function () {
+      var cartocss = '#layer { polygon-fill: white; polygon-opacity: 0.1; ::outline { line-width: 0.5; line-opacity: 0.7; }}';
+      var attr = 'line-opacity';
+      var newStyle = 0.2;
+
+      expect(StyleUtils.changeStyle(cartocss, attr, newStyle)).toBe(cartocss);
+    });
+
+    it('should not replace line-color if it is included under a layer::outline symbolizer', function () {
+      var cartocss = '#layer { polygon-fill: white; polygon-opacity: 0.1; } #layer::outline { line-width: 0.5; line-opacity: 0.7; }';
+      var attr = 'line-opacity';
+      var newStyle = 0.2;
+
+      expect(StyleUtils.changeStyle(cartocss, attr, newStyle)).toBe(cartocss);
+    });
+
+    it('should not replace the style if it\'s undefined', function () {
+      var cartocss = '#layer { polygon-fill: white; polygon-opacity: 0.1; }';
+      var attr = 'polygon-opacity';
+
+      expect(StyleUtils.changeStyle(cartocss, attr, undefined)).toBe(cartocss);
+    });
   });
```

Comment:
It won't do anything, if the value you're trying to set is undefined, it will return the current css without changing it